### PR TITLE
Fix function returns and enable more expression tests

### PR DESCRIPTION
**Pull request by jeanPerier**
- fix function distance to support fir.real vs mlir.float (require to search runtime for intrinsics)
- fix function returns when no explicit parser::ReturnStmt (regression from change to "ast")
- enable expression lowering tests with complex

The expression tests (scalar intrinsic operations: ~600 expression variants) now all passes with the following change (not part of this commit) :
- rebase with latest front-end (containing pr 865)
- mangle external function names like pgfortran (i.e `foo` -> `foo_`). The internal mangling `_QPfoo` is currently going until code generation and that prevent correct linking with pgfortran compiled driver.

### CODE REVIEW

File: lib/burnside/bridge.cc [hunk 1]
```diff
@@ -94,6 +94,10 @@ class FirConverter : public AbstractConverter {
     return createI1LogicalExpression(
         loc, *builder, *expr, localSymbols, defaults, intrinsics);
   }
+  M::Value *createTemporary(M::Location loc, const Se::Symbol &sym) {
+    return ::createTemporary(loc, *builder, localSymbols,
```
Comment (schweitzpgi) [inline lib/burnside/bridge.cc]: Is this function being called not in a namespace?

File: lib/burnside/intrinsics.cc [hunk 2]
```diff
@@ -329,7 +334,20 @@ class FunctionDistance {
...
+    // - use fir/KindMapping.h and LLVM::Type
+    // - or use evaluate/type.h
+    if (auto r{t.dyn_cast<fir::RealType>()}) return r.getFKind() * 4;
+    if (auto cplx{t.dyn_cast<fir::CplxType>()}) return cplx.getFKind() * 4;
+    assert(false && "not a floating-point type");
+    return 0;
+  }
+  static bool isFloatingPointType(mlir::Type t) {
+    return t.isa<mlir::FloatType>() || t.isa<fir::RealType>();
+  }
```
Comment (schweitzpgi) [inline lib/burnside/intrinsics.cc]: There should already be something like this that gets built from the .td file ... but, I can't recall what it is called off the top of my head.  Bitwidths and subtype testing, I mean.

Comment (jeanPerier) [reply lib/burnside/intrinsics.cc]: Mmmh, you mean something generated from the the `AnyRealLike` constraint ?
I looked into generated `tools/flang/include/fir/FIROps.cpp.inc` and `tools/flang/include/fir/FIROps.h.inc` but could not find anything like that.

Looking at `AddfOp::verify` in `FIROps.cpp.inc` (currently line 284), it looks like the `AnyRealLike` part has been directly inlined into `((((v->getType().isa<FloatType>())) || (((v->getType().isa<VectorType>())) && ((v->getType().cast<ShapedType>().getElementType().isa<FloatType>()))) || (((v->getType().isa<TensorType>())) && ((v->getType().cast<ShapedType>().getElementType().isa<FloatType>())))) || ((v->getType().isa<fir::RealType>())))` (yiikes, more complex than my test as it seems to account for vector/tensor types when I only care about scalar).

File: lib/burnside/bridge.cc (hunk 1 above)
Comment (jeanPerier) [reply lib/burnside/bridge.cc]: I saw you have already changed this to `Br::` thanks, that is clearer than `::`.

File: lib/burnside/intrinsics.cc (hunk 2 above)
Comment (schweitzpgi) [reply lib/burnside/intrinsics.cc]: Fair enough. Thanks for digging around.

**Pull request merged**